### PR TITLE
fix(table): fix flaky detaching of documents when updating TableAnswers

### PR DESCRIPTION
**Pull request by czosel**
Under certain conditions (which are not fully understood until now) the
main documents' family was changed when an enclosed table question was
updated.

This adds a test that demonstrates the flaky behaviour, and a bugfix
that fixes it and tries to clarify the code.

### CODE REVIEW

**Comment by czosel** [comment]
@sliverc I just had another look - yesterday's fix actually seems fine to me. We got confused because we were calling the return value of `_get_document_tree` `children`, but it also returns the document ID that was passed to it (`yield document_id`, `L545`). I renamed the variable to `tree` and added a comment for clarification.

I'll test this in one of our projects now, afterwards this should be good to go from my side.

**Comment by czosel** [comment]
Test in customer project was successful :+1:

File: caluma/form/serializers.py [hunk 1]
```diff
@@ -534,12 +534,16 @@ class SaveDocumentTableAnswerSerializer(SaveAnswerSerializer):
     )
 
     def _get_document_tree(self, document_id):
-        answers = models.AnswerDocument.objects.filter(document_id=document_id).values(
-            "answer"
+        """Get all child documents (table rows) of a given document, and the document itself."""
+
+        answers = models.Answer.objects.filter(document_id=document_id)
```
Comment (sliverc) [inline caluma/form/serializers.py]: better to be filter after type `TYPE_TABLE` as well? Could then call it table_answers to make the code better understood.

File: caluma/form/serializers.py [hunk 2]
```diff
@@ -534,12 +534,16 @@ class SaveDocumentTableAnswerSerializer(SaveAnswerSerializer):
...
+        child_documents = (
+            answers.filter(documents__isnull=False)
```
Comment (sliverc) [inline caluma/form/serializers.py]: I have a feeling isnull on a m2m is always false... with change above this would not really be needed though.

**Pull request merged**